explorer-ui/src/components/MatchGrid/AllUsers/AllUsers.jsx

```diff
@@ -31,7 +31,7 @@ export default function AllUsers({profiles}) {
     const getCoordinates = (profiles) => {
         if (profiles) {
           for (let i = 0; i < profiles.length; i++) {
-            if(profiles[i].address != null || profiles[i].address!="" ){
+            if(( profiles[i].address != null || profiles[i].address!="") && profiles[i].profileInfo.visibility =="yes"){
             Geocode.fromAddress(profiles[i].address).then(
```

Comment:
as this is a promise -> need to have it blocking, otherwise the for loop will execute them all in parallel leading to unpredictable behavior

Reply:
fixed